providers/amazon/src/airflow/providers/amazon/aws/hooks/sagemaker_unified_studio.py

```diff
@@ -0,0 +1,188 @@
+# under the License.
+
+"""This module contains the Amazon SageMaker Unified Studio Notebook hook."""
+
+import time
+
+from sagemaker_studio import ClientConfig
+from sagemaker_studio._openapi.models import GetExecutionRequest, StartExecutionRequest
+from sagemaker_studio.sagemaker_studio_api import SageMakerStudioAPI
+
+from airflow import AirflowException
+from airflow.hooks.base import BaseHook
+from airflow.providers.amazon.aws.utils.sagemaker_unified_studio import is_local_runner
+
+
+class SageMakerNotebookHook(BaseHook):
+    """
+    Interact with Sagemaker Unified Studio Workflows.
+
+    This hook provides a wrapper around the Sagemaker Workflows Notebook Execution API.
+
+    Examples:
+     .. code-block:: python
+
+        from airflow.providers.amazon.aws.hooks.sagemaker_unified_studio import SageMakerNotebookHook
+
+        notebook_hook = SageMakerNotebookHook(
+            input_config={'input_path': 'path/to/notebook.ipynb', 'input_params': {'param1': 'value1'}},
+            output_config={'output_uri': 'folder/output/location/prefix', 'output_formats': 'NOTEBOOK'},
+            execution_name='notebook_execution',
+            waiter_delay=10,
+            waiter_max_attempts=1440,
+        )
+
+    :param execution_name: The name of the notebook job to be executed, this is same as task_id.
+    :param input_config: Configuration for the input file.
+        Example: {'input_path': 'folder/input/notebook.ipynb', 'input_params': {'param1': 'value1'}}
+    :param output_config: Configuration for the output format. It should include an output_formats parameter to specify the output format.
+        Example: {'output_formats': ['NOTEBOOK']}
+    :param compute: compute configuration to use for the notebook execution. This is a required attribute
+        if the execution is on a remote compute.
+        Example: { "InstanceType": "ml.m5.large", "VolumeSizeInGB": 30, "VolumeKmsKeyId": "", "ImageUri": "string", "ContainerEntrypoint": [ "string" ]}
+    :param termination_condition: conditions to match to terminate the remote execution.
+        Example: { "MaxRuntimeInSeconds": 3600 }
+    :param tags: tags to be associated with the remote execution runs.
+        Example: { "md_analytics": "logs" }
+    :param waiter_delay: Interval in seconds to check the task execution status.
+    :param waiter_max_attempts: Number of attempts to wait before returning FAILED.
+    """
+
+    def __init__(
+        self,
+        execution_name: str,
+        input_config: dict = {},
+        output_config: dict = {"output_formats": ["NOTEBOOK"]},
+        compute: dict = {},
+        termination_condition: dict = {},
+        tags: dict = {},
+        waiter_delay: int = 10,
+        waiter_max_attempts: int = 1440,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self._sagemaker_studio = SageMakerStudioAPI(self._get_sagemaker_studio_config())
```

Comment:
Does SageMakerStudioAPI invoke an API call to SageMaker? If so I am not sure if it's advisable to have it in the hook init function?

Reply:
It doesn't make an API call to SageMaker, but it creates the client for the SageMakerStudioSDK that we'll use to interact with the SageMaker API in this hook.

Just for my knowledge, what's the issue with making an API call to SageMaker in the init?

Reply:
If there are no api calls there arent any.
Hooks are used with custom operators. Some users use it in the init function of the operator which may cause problems.

Reply:
Happy to make this change in a fix, but I don't think we need to block this one because this Hook is only used in the `SagemakerNotebookOperator` as of now and isn't designed to be used in any other custom Operators.